torch/utils/tensorboard/pytorch_graph.py

```diff
@@ -0,0 +1,274 @@
+import numpy as np
+import time
+import torch
+
+from collections import OrderedDict
+from distutils.version import LooseVersion
+
+from tensorboard.compat.proto.config_pb2 import RunMetadata
+from tensorboard.compat.proto.graph_pb2 import GraphDef
+from tensorboard.compat.proto.step_stats_pb2 import StepStats, DeviceStepStats, NodeExecStats, AllocatorMemoryUsed
+from tensorboard.compat.proto.versions_pb2 import VersionDef
+
+from .proto_graph import Node_proto
+
+
+methods_OP = ['attributeNames', 'hasMultipleOutputs', 'hasUses', 'inputs',
+              'kind', 'outputs', 'outputsSize', 'scopeName']
+methods_IO = ['node', 'offset', 'uniqueName']  # 'unique' <int> , 'type' <Tensor<class 'torch._C.Type'>>
+
+
+class Node_base(object):
+    def __init__(self, uniqueName=None, inputs=None, scope=None, tensorSize=None, op_type='UnSpecified', attributes=''):
+        self.uniqueName = uniqueName
+        self.inputs = inputs
+        self.tensorSize = tensorSize
+        self.kind = op_type
+        self.attributes = attributes
+        if scope is not None:
+            self.scope = scope
+
+    def __repr__(self):
+        repr = []
+        repr.append(str(type(self)))
+        for m in dir(self):
+            if '__' not in m:
+                repr.append(m + ': ' + str(getattr(self, m)) + str(type(getattr(self, m))))
+        return '\n'.join(repr) + '\n\n'
```

Comment:
It's too bad we can't use https://docs.python.org/3/library/dataclasses.html in PyTorch, this would have eliminated all the boilerplate here.

Reply:
__repr__ here is for debug `pytorch_graph.py`. Can we do it later?

Reply:
This comment isn't actionable :) So yes, there is nothing to do, you can do it later ;)